docs/guides/circuits.rst

```diff
@@ -0,0 +1,439 @@
+
+The output we get from those two ``icecream`` statements should be
+
+.. code-block:: text
+
+  ic| unitary.shape: (8, 8)
+  ic| type(unitary): <class 'numpy.ndarray'>
+
+Inverting a circuit
+-------------------
+
+In some contexts, you may want to invert a circuit. For instance, if instead of :ref:`building a bell state <creating_basic_circuits>`, you want to perform a bell measurement. Or perhaps your circuit requires some `uncomputation <https://quantum.country/search#building-blocks-quantum-search>`_ that you don't want to build manually. To get the inverse of a circuit, just call the ``.inverse()`` method on it:
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: = state_prep_circ.inverse()
+  :end-at: ic(np.allclose(inverse_unitary, np.eye(8)))
+
+In the output we can see that, indeed, the Hadamard gate is its own inverse and when we append this circuit to the ``state_prep_circ`` it does in fact invert it, leaving us with the identity matrix:
+
+.. code-block:: text
+
+  ic| invert_state_prep_circ: Circuit(operations=[H(2), H(1), H(0)], n_qubits=3)
+  ic| np.allclose(inverse_unitary, np.eye(8)): True
+
+
+.. _appending_circuits:
+
+Appending operations to a circuit
+---------------------------------
+
+You don't have to build a whole circuit all at once! Let's see how to append operations to an existing circuit by constructing the next part of our QAOA circuit, the problem hamiltonian circuit. This problem hamiltonian circuit gives a quantum description of the problem and shows how the nodes are connected. In this case, the 3 nodes are all-to-all connected, so our circuit should look like this:
+
+.. code-block:: text                                                                     
+                                                                                
+  q_0: ──■───────────────────■────■───────────────────■───────────────────────────
+       ┌─┴─┐┌─────────────┐┌─┴─┐  │                   │                           
+  q_1: ┤ X ├┤ Rz($\beta$) ├┤ X ├──┼───────────────────┼────■───────────────────■──
+       └───┘└─────────────┘└───┘┌─┴─┐┌─────────────┐┌─┴─┐┌─┴─┐┌─────────────┐┌─┴─┐
+  q_2: ─────────────────────────┤ X ├┤ Rz($\beta$) ├┤ X ├┤ X ├┤ Rz($\beta$) ├┤ X ├
+                                └───┘└─────────────┘└───┘└───┘└─────────────┘└───┘
+
+We'll build up this circuit one connection at a time, using ``+=`` to append ``Circuit`` objects to the end of our existing ``problem_hamiltonian_circ``
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: problem_hamiltonian_circ = Circuit(
+  :end-before: unitary = problem_hamiltonian_circ.to_unitary()
+
+For now, don't worry about what the ``beta`` variables mean in these circuits. We'll cover that later when we talk about :ref:`symbolic gates <symbolic_gates>`. For now, just notice that we started with an initial circuit and we were able to append other circuits to it using ``+=``
```

Comment:
I'll put the first sentence before the circuit. Cause I immediately started worrying where `beta` is defined :D

Reply:
fixed in af446a1749110a289630084291990551053306a0